presto-main/src/main/java/com/facebook/presto/sql/gen/ExpressionCompiler.java

```diff
@@ -90,13 +93,16 @@ public PageProcessor compilePageProcessor(RowExpression filter, List<RowExpressi
     private <T> T compileAndInstantiate(RowExpression filter, List<RowExpression> projections, BodyCompiler<T> bodyCompiler, Class<? extends T> superType)
     {
         // create filter and project page iterator class
-        Class<? extends T> clazz = compileProcessor(filter, projections, bodyCompiler, superType);
         try {
+            Class<? extends T> clazz = compileProcessor(filter, projections, bodyCompiler, superType);
             return clazz.newInstance();
         }
         catch (ReflectiveOperationException e) {
             throw Throwables.propagate(e);
         }
+        catch (CompilationException e) {
+            throw new PrestoException(COMPILER_ERROR, e.getMessage(), e.getCause());
```

Comment:
Make this `new PrestoException(COMPILER_ERROR, e.getCause())`. We should actually fix `PrestoException` to use the message from the cause when one is not supplied